File: Project_Go_Stack_by_Array/stackspeedtest/stackspeedtest.go
package solution

import (
	"fmt"
	"strconv"
	"time"

	stack "./stack"
)

func Main(max int) {
	var stk *stack.Stack

	timeStart := time.Now()

	stk = stack.Constructor()

	for i := 1; i <= max; i++ {
		stk.Push(i)
//		fmt.Printf("stk.Push(%d)\n", i)
	}

	fmt.Printf("stk.Push(%d) done.\n", max)
	fmt.Printf("stk.Search(%d) ... %d\n", 1, stk.Search(1))

	for stk.Empty() == false {
/*
		size    := stk.Size()
		peek, _ := stk.Peek()
		pop , _ := stk.Pop()
		fmt.Printf("size = %d, peek = %d, pop = %d\n", size, peek, pop)
*/
		stk.Pop()

	}

	fmt.Printf("stk.Pop() done.\n")
	fmt.Printf("stk.Empty()   ... %s\n", strconv.FormatBool(stk.Empty()))

	timeEnd := time.Now()

	fmt.Printf("Execute time : %.3f [ms]\n\n", timeEnd.Sub(timeStart).Seconds()*1000)
}

File: Project_C_Stack_by_ListNode/lib/stackspeedtest.c
#include <time.h>
#include "../include/stackspeedtest.h"

// #define DEBUG

int stackspeedtest(int max) {
    clock_t time_start = clock();
    Stack *s = stackCreate();

/*
    stackPush(s, 1);
    stackPush(s, 2);
    stackPush(s, 3);
    printf("pop = %d\n", stackPop(s));
    printf("pop = %d\n", stackPop(s));
    printf("pop = %d\n", stackPop(s));
*/

    for (int i = 1; i <= max; i++) {
        stackPush(s, i);
#ifdef DEBUG
        printf("stackPush(%d)\n", i);
#endif
    }

    printf("stackPush(%d) done.\n", max);
//  printf("stackSearch(1) ... %d\n", stackSearch(s, 1));
    printf("stackSearch(%d) ... %d\n", max, stackSearch(s, max));

    while (stackEmpty(s) == false) {

#ifdef DEBUG
        int size = stackSize(s);
        int top  = stackTop(s);
        int pop  = stackPop(s);
        printf("size = %d, top = %d, pop = %d\n", size, top, pop);
#else
        stackPop(s);
#endif

    }

    printf("stackPop() done.\n");
    if (stackEmpty(s) != 0) {
        printf("stackEmpty() ... %s\n", "true");
    } else {
        printf("stackEmpty() ... %s\n", "false");
    }

    printf("destroy stack.\n");
    stackDestroy(s);

    clock_t time_end = clock();
    printf("Execute time : %f [ms]\n\n", (double)(time_end - time_start)*1000/CLOCKS_PER_SEC);

    return 0;
}
